<?xml version="1.0"?>
<launch>
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find sia_arm_description)/urdf/sia_robot_visual_fix.urdf.xacro'" />

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find sia_arm_description)/test/urdf_test.rviz" required="true" />

    <arg name="num_samples" default="1000" />
    <arg name="chain_start" default="link1" />
    <!--<arg name="chain_end" default="tool" />-->
    <arg name="chain_end" default="eef" />
    <arg name="timeout" default="0.005" />

    <node name="trac_ik_tests" pkg="trac_ik_examples" type="sia_arm_test" output="screen">
        <param name="num_samples" value="$(arg num_samples)"/>
        <param name="chain_start" value="$(arg chain_start)"/>
        <param name="chain_end" value="$(arg chain_end)"/>
        <param name="timeout" value="$(arg timeout)"/>
        <param name="urdf_param" value="/robot_description"/>
    </node>

</launch>